### ISSUE TITLE
ignore case for IQueryable

### ISSUE THREAD

**Issue by veezed** [ninjanye/SearchExtensions#15]
Some databases, such as Postgre is case sensitive.

You could implement case insensitive queries for IQueryable, by using toLower before applying filter clause. 
I implemented it on my own branch on 'contains', I could share it here if you want.

There is small example:
![image](https://cloud.githubusercontent.com/assets/4472342/20917441/3b3d9dce-bb9a-11e6-920c-935f5a156fd3.png)

![image](https://cloud.githubusercontent.com/assets/4472342/20917450/4ba76d8e-bb9a-11e6-96c1-e58b6aa53015.png)

**Comment by ninjanye** [comment]
This is certainly a good new feature.  Be good to have it as a configurable option though.  e.g.

    context.Search(x => x.String1).CaseInsensitive().Contains("TeSt");

I'll can try and look at this in more detail and get something pushed through.

**Comment by veezed** [comment]
Yes, I made it in my branch, it just have different name. In my project it's called IgnoreCase()

**Comment by ninjanye** [comment]
That sounds good.  Have you implemented it for all comparison methods (e.g. `.StartsWith`, `.Contains`, `.EndsWith` etc?) or just the contains method?  If it's all done, feel free to submit a PR.

**Comment by veezed** [comment]
Sorry, just Contains

**Comment by PatryxCShark** [comment]
Hi! Where is a solution because I cannot find it?
I want to have possibility to decide about case sensitive or case insensitive comparison especially when it comes to Containing.
Thanks in advanced

**Comment by veezed** [comment]
What Db engine do you use?

**Comment by PatryxCShark** [comment]
MS SQL, but sometimes I have IQueryable which is done in memory. If I know is process on memory side, should I do IQueryable.AsEnumerable() and then use SetCulture?

**Comment by veezed** [comment]
Ms sql is case insensitive. This change is required only for DB engines who
are case sensitive, for example postgres sql.
